app/src/androidTest/java/com/example/samplegallery/PhotoBlowupFragmentTest.java

```diff
@@ -0,0 +1,140 @@
+        VolleyQueueIdlingResource vqidr = new VolleyQueueIdlingResource("Volley Queue");
+        VolleyRequestQueue.getInstance(null).getRequestQueue().start();
+        Espresso.registerIdlingResources(vqidr);
+
+        // navigate to an album
+        onData(anything())
+                .inAdapterView(withId(R.id.albums_list_view))
+                .atPosition(0)
+                .perform(click());
+
+        // attempt to click on a photo to blow it up
+        onData(anything())
+                .inAdapterView(withId(R.id.gridview))
+                .atPosition(0)
+                .perform(click());
+
+        // the description, title, and photo itself should all be visible together
+        onView(withId(R.id.photo_blowup))
+                .check(matches(isDisplayed()));
+        onView(withResourceName("photo_description"))
+                .check(matches(isDisplayed()));
+        onView(withResourceName("photo_title"))
+                .check(matches(isDisplayed()));
+        // and the progress bar should be hidden
+        onView(withId(R.id.progress_bar_blowup))
+                .check(matches(not(isDisplayed())));
+    }
+
+    private class VolleyQueueIdlingResource implements IdlingResource {
+        private Field mCurrentRequests;
+        private String resName;
+        private volatile ResourceCallback resourceCallback;
+
+        VolleyQueueIdlingResource(String resourceName) throws NoSuchFieldException {
+            resName = resourceName;
+            mCurrentRequests = RequestQueue.class.getDeclaredField("mCurrentRequests");
+            mCurrentRequests.setAccessible(true);
+        }
+
+        @Override
+        public String getName() {
+            return resName;
+        }
+
+        @Override
+        public boolean isIdleNow() {
+            try {
+                synchronized (mCurrentRequests.get(
+                        VolleyRequestQueue.getInstance(null).getRequestQueue()
+                )) {
+                    @SuppressWarnings("unchecked")
+                    Set<Request> requests = (Set<Request>) mCurrentRequests.get(
+                            VolleyRequestQueue.getInstance(null).getRequestQueue()
+                    );
+
+                    if (requests.size() == 0) {
+                        if(resourceCallback != null) {
+                            resourceCallback.onTransitionToIdle();
+                        }
+
+                        return true;
+                    }
+
+                    return false;
+                }
+            } catch (IllegalAccessException e) {
+                Log.e("AlbumsFragmentTest", "Cannot access request queue variable!");
+                e.printStackTrace();
+                System.exit(-1);
+            }
+
+            return false;
+        }
+
+        @Override
+        public void registerIdleTransitionCallback(ResourceCallback callback) {
+            resourceCallback = callback;
+        }
+    }
+}
```

Comment:
Definitely a good start. That said, there are a few things your should think about:

1) There are ways to test a fragment "in isolation". That is, you should make an activity which loads the fragment directly (not by navigation), since we would like to only assert that a fragment works as it is supposed, not that the fragment interacts with the application the way we want to (that should be another test!). 

http://stackoverflow.com/questions/8199000/unit-test-an-android-fragment

The above describes what I mean in more detail. I will be looking into modularizing such tests within our actual app, but for now you should just follow what is explained there!

2) Before you start writing tests, write out what exactly must happen, step-by-step. That is, include such things like order of request completion (what happens if we get the description before the photo? What happens if we get the photo before the description? etc.)

Overall, good start! :-)